### PR TITLE
docs: Extend password entry of ansible.builtin.user

### PR DESCRIPTION
**Pull request by Hofer-Julian**
##### SUMMARY
Clarify that `password` sets the password hash and not the actual password.
Fixes part of  #79684

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
ansible.builtin.user

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
@samccann

File: lib/ansible/modules/user.py
Comment (sivel) [inline lib/ansible/modules/user.py]: This may cause confusion with the statement lower, that says that you are supposed to do this for macOS.  So perhaps this statement and the `macOS` statement should be combined.

Comment (Hofer-Julian) [reply lib/ansible/modules/user.py]: I was confused by the macOS sentence too.
So on macOS the password field expects the actual password, on Linux the hash?
What is expected on Windows?

Comment (bcoca) [reply lib/ansible/modules/user.py]: to use win_user instead

Comment (Hofer-Julian) [reply lib/ansible/modules/user.py]: I gave the phrasing another try.

Regarding the differing behavior of Linux and macOS: I wonder if I should open an issue to discuss this behavior. Using the actual password on one platform and the hash on another one sounds like a foot gun to me.

File: lib/ansible/modules/user.py
Comment (oraNod) [inline lib/ansible/modules/user.py]:
```suggestion
            - Optionally set the user password.
```

File: lib/ansible/modules/user.py
Comment (oraNod) [inline lib/ansible/modules/user.py]:
```suggestion
            - B(Linux:) Enter the hashed password as the value.
```

File: lib/ansible/modules/user.py
Comment (oraNod) [inline lib/ansible/modules/user.py]:
```suggestion
            - B(macOS:) Enter the cleartext password as the value. Be sure to take relevant security precautions.
```

**Review by oraNod — COMMENTED** [review]
Adding some suggestions to try and clarify the choices a little more, if I understand this correctly.

I would prefer more authoritative language that instructs users to enter exactly what each OS requires. That might get rid of some of the ambiguity/confusion. But it does seem concerning that macOS doesn't take a hashed password.

File: lib/ansible/modules/user.py
Comment (bcoca) [inline lib/ansible/modules/user.py]: - If provided, set the user's password to the provided encrypted hash (linux/unix/POSIX) or plain text password (OS X/Mac OS)

Comment (Hofer-Julian) [reply lib/ansible/modules/user.py]: Isn't that redundant with the text below?

Also, I don't think that Linux is "more Unix" than macOS :)

Comment (bcoca) [reply lib/ansible/modules/user.py]: why its in list of linux/unix/posix otherwise it woudl be implied under unix

Comment (bcoca) [reply lib/ansible/modules/user.py]: and yes, it makes other lines redundant, but i think its simple, concise and clear in one sentence and will avoid people 'trailing off' and not reading a wall of text.

**Comment by samccann** [comment]
Hi @Hofer-Julian  - have you had a chance to consider the feedback above?  Thanks for your help!

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
LGTM but let's see if @bcoca has any final comments before we merge.

File: lib/ansible/modules/user.py
Comment (bcoca) [inline lib/ansible/modules/user.py]: it's not just linux, but most POSIXy/unixy systems (i know Mac OS is BSD derived but here it deviates from the norm).

Comment (Hofer-Julian) [reply lib/ansible/modules/user.py]: Feel free to push to this branch with the wording you like.

Comment (samccann) [inline lib/ansible/modules/user.py]:
```suggestion
            - B(Linux/Unix/POSIX:) Enter the hashed password as the value.
```

File: lib/ansible/modules/user.py
Comment (samccann) [inline lib/ansible/modules/user.py]:
```suggestion
            - B(OS X/macOS:) Enter the cleartext password as the value. Be sure to take relevant security precautions.
```

**Review by samccann — APPROVED** [review]

**Pull request merged**
